pkg/entrypoint/runner.go

```diff
@@ -180,9 +193,93 @@ func (rr *RealRunner) scan(ctx context.Context, mu *url.URL, scanner *bufio.Scan
 	done <- true
 }
 
-func updatePath() error {
-	currentPath := os.Getenv("PATH")
-	return os.Setenv("PATH", currentPath+":/var/lib/puppet/relay")
+func (rr *RealRunner) handleProcessState(ctx context.Context, mu *url.URL, ps *os.ProcessState, wcs model.WhenConditionStatus) {
+	if ps != nil && mu != nil {
+		_ = rr.putStatus(ctx, mu,
+			&model.ActionStatus{
+				ProcessState: &model.ActionStatusProcessState{
+					ExitCode: ps.ExitCode(),
+				},
+				WhenCondition: &model.ActionStatusWhenCondition{
+					WhenConditionStatus: wcs,
+				},
+			},
+		)
+	}
+}
+
+func (rr *RealRunner) processWhenConditions(ctx context.Context, mu *url.URL) (model.WhenConditionStatus, error) {
+	_ = rr.putStatus(ctx, mu,
+		&model.ActionStatus{
+			WhenCondition: &model.ActionStatusWhenCondition{
+				WhenConditionStatus: model.WhenConditionStatusEvaluating,
+			},
+		})
+
+	whenConditionStatus, err := rr.evaluateConditions(ctx, mu)
+
+	_ = rr.putStatus(ctx, mu,
+		&model.ActionStatus{
+			WhenCondition: &model.ActionStatusWhenCondition{
+				WhenConditionStatus: whenConditionStatus,
+			},
+		})
+
+	return whenConditionStatus, err
+}
+
+func (rr *RealRunner) evaluateConditions(ctx context.Context, mu *url.URL) (model.WhenConditionStatus, error) {
+	ee := &url.URL{Path: "/conditions"}
+
+	req, err := http.NewRequestWithContext(
+		ctx, http.MethodGet, mu.ResolveReference(ee).String(), http.NoBody)
+	if err != nil {
+		return model.WhenConditionStatusFailure, err
+	}
+
+	// TODO This needs to be configurable
+	contextWithTimeout, cancel := context.WithTimeout(ctx, 60*time.Minute)
+	defer cancel()
+
+	success := false
+	err = retry.Wait(contextWithTimeout, func(ctx context.Context) (bool, error) {
+		resp, err := rr.getResponse(ctx, req, []retry.WaitOption{})
+		if err != nil {
+			return retry.Done(err)
+		}
+		defer resp.Body.Close()
+
+		switch resp.StatusCode {
+		case http.StatusOK:
+			var env api.GetConditionsResponseEnvelope
+			if err := json.NewDecoder(resp.Body).Decode(&env); err != nil {
+				if err == io.EOF {
```

Comment:
Is this another temporary error? I don't think I've ever seen this in a JSON decode check before.

Reply:
I believe this is only technically in place for (one?) test that is artificially responding to the `/conditions` endpoint with a 200 and not sending any payload. I don't believe the actual endpoint can naturally cause this situation, but it's not a bad safety check either in this case.

Either way, as soon as this code is in place, we can rewrite the `/conditions` endpoint to remove this custom response envelope (or make other similar improvements), as it was really only necessary to support the legacy scripting (with relatively simple control mechanisms to avoid increasingly complex error parsing).

Reply:
All of that should be easily refactored now that this is properly codified.